scripts/clean-shrinkwrap.js

```diff
@@ -1,43 +1,166 @@
- * dependendencies are platform dependent and you then try to build
- * the project in another platform.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
  *
- * As a workaround, we keep a list of platform dependent dependencies in
- * the `shrinkwrapIgnore` property of `package.json`, and manually remove
- * them from `npm-shrinkwrap.json` if they exists.
+ *    http://www.apache.org/licenses/LICENSE-2.0
  *
- * See: https://github.com/npm/npm/issues/2679
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This script is in charge of removing platform specific optional
+ * dependencies, and their dependencies from the `shrinkwrap` file.
  */
 
 'use strict';
 
+const _ = require('lodash');
+const fs = require('fs');
 const path = require('path');
-const os = require('os');
 const packageJSON = require('../package.json');
 const spawn = require('child_process').spawn;
-const shrinkwrapIgnore = packageJSON.shrinkwrapIgnore;
+const childProcess = require('child_process');
 
-console.log('Removing:', shrinkwrapIgnore.join(', '));
+const NPM_SHRINKWRAP_FILE_PATH = path.join(__dirname, '..', 'npm-shrinkwrap.json');
+const shrinkwrapFile = require(NPM_SHRINKWRAP_FILE_PATH);
+const platformSpecificDependencies = require('../package.json').platformSpecificDependencies;
 
 /**
- * Run an npm command
- * @param {Array} command - list of arguments
- * @returns {ChildProcess}
+ * @summary Normalize npm qualifiers
+ * @function
+ * @private
+ *
+ * @param {String[]} qualifiers - npm qualifiers
+ * @returns {String[]} normalized qualifiers
+ *
+ * @example
+ * const qualifiers = normalizeNpmQualifiers([ 'drivelist@^5.0.0', 'electron@1.6.6' ]);
+ * console.log(qualifiers);
+ * > [ 'drivelist', 'electron' ]
  */
-const npm = (command) => {
-  const npmBinary = os.platform() === 'win32' ? 'npm.cmd' : 'npm';
-  return spawn(npmBinary, command, {
-    cwd: path.join(__dirname, '..'),
-    env: process.env,
-    stdio: [ process.stdin, process.stdout, process.stderr ]
+const normalizeNpmQualifiers = (qualifiers) => {
+  return _.map(qualifiers, (qualifier) => {
+    return _.first(_.split(qualifier, '@'));
   });
 };
 
-npm([ 'rm', '--ignore-scripts' ].concat(shrinkwrapIgnore))
-  .once('close', () => {
-    console.log('Done.');
+/**
+ * @summary Get the top level dependencies of a certain npm specifier
+ * @function
+ * @private
+ *
+ * @param {String} specifier - npm specifier
+ * @returns {String[]} top level dependencies
+ *
+ * @example
+ * const dependencies = getTopLevelDependencies('drivelist@^5.0.0');
```

Comment:
Does this need to be modified to fetch details about a specific version, rather than a 'version range' (or whatever the correct terminology is) ?
Compare:
```
$ npm view drivelist@^5.0.0 dependencies
drivelist@5.0.0 { 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.1 { 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.2 { 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.3 { 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.4 { 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.5 { 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.6 { 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.7 { 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.8 { 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.9 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.10 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.11 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.12 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.13 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.14 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.15 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.16 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.17 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.18 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.19 { debug: '^2.6.0', 'js-yaml': '^3.4.1', lodash: '^4.16.4' }
drivelist@5.0.20  
{ bluebird: '^3.5.0',
  debug: '^2.6.0',
  'js-yaml': '^3.4.1',
  lodash: '^4.16.4',
  tmp: '0.0.31' }
drivelist@5.0.21  
{ bluebird: '^3.5.0',
  debug: '^2.6.0',
  'js-yaml': '^3.4.1',
  lodash: '^4.16.4',
  tmp: '0.0.31' }
drivelist@5.0.22  
{ bluebird: '^3.5.0',
  debug: '^2.6.0',
  'js-yaml': '^3.4.1',
  lodash: '^4.16.4',
  tmp: '0.0.31' }
```
with:
```
$ npm view drivelist@5.0.22 dependencies
{ bluebird: '^3.5.0',
  debug: '^2.6.0',
  'js-yaml': '^3.4.1',
  lodash: '^4.16.4',
  tmp: '0.0.31' }
```

Obviously it's possible that different versions of a module might have distinct dependencies, and we only want the dependency-list for the _actual_ version of the module we're using?
E.g. `somemodule@1.0.2` might depend on `['othermoduleA', 'othermoduleB']`, `somemodule@1.0.3` might depend on `['othermoduleB', 'othermoduleC']` and `somemodule@1.0.4` might depend on `['othermoduleC', 'othermoduleD']`; and if we happen to be using `somemodule@1.0.3`, then when we ask for `somemodule`'s dependencies, we want to be told `['othermoduleB', 'othermoduleC']` and not  `['othermoduleA', 'othermoduleB', 'othermoduleC', 'othermoduleD']`. (if that makes sense?)

But for modules that aren't shrinkwrapped, I've no idea how difficult it is to find out the actual version we're using, so feel free to ignore me if I'm just overcomplicating things ;-)